pdns/pdns_recursor.cc

```diff
@@ -1006,7 +1006,8 @@ void startDoResolve(void *p)
 
       bool needCommit = false;
       for(auto i=ret.cbegin(); i!=ret.cend(); ++i) {
-        if(!DNSSECOK && (i->d_type == QType::RRSIG || i->d_type==QType::NSEC || i->d_type==QType::NSEC3))
+        if(!DNSSECOK && ((((i->d_type == QType::RRSIG || i->d_type==QType::NSEC) && i->d_place != DNSResourceRecord::ANSWER) || i->d_type==QType::NSEC3) ||
+                        (i->d_type == QType::RRSIG && dc->d_mdp.d_qtype != i->d_type &&  dc->d_mdp.d_qtype != QType::ANY)))
```

Comment:
Can we reformat this to more lines so the () levels are clearer? And can you replace the last `i->d_type` with `QType::RRSIG`? I think this would make the code easier to understand.

Otherwise LGTM.